llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -361,14 +361,22 @@ attributeToExecModeMetadata(const Attribute &Attr, Function &F) {
       AddFPControlMetadataForWidth(SPIRV_DENORM_PRESERVE, 64);
   }
 
-  if (AttrKindStr == "sycl-work-group-size" ||
-      AttrKindStr == "sycl-work-group-size-hint") {
+  static constexpr std::pair<const char *, const char *> SimpleWGAttrs[] = {
+      {"sycl-work-group-size", "reqd_work_group_size"},
+      {"sycl-work-group-size-hint", "work_group_size_hint"},
+      {"sycl-max-work-group-size", "max_work_group_size"},
+  };
+
+  for (auto [AttrKind, MDStr] : SimpleWGAttrs) {
+    if (AttrKindStr != AttrKind)
+      continue;
     // Split values in the comma-separated list integers.
     SmallVector<StringRef, 3> ValStrs;
     Attr.getValueAsString().split(ValStrs, ',');
 
     assert(ValStrs.size() <= 3 &&
-           "sycl-work-group-size and sycl-work-group-size-hint currently only "
+           "sycl-work-group-size, sycl-work-group-size-hint and "
+           "sycl-max-work-group-size currently only "
```

Comment:
Looks like the NVPTX changes above require all 3 values to be there. Should this also be checking that all 3 are there?

Reply:
Yes, it should. Let me see how best to do that while reusing as much code as possible.

Reply:
Done!